### ISSUE TITLE
Limel-button-group onClick event triggers twice & doesn't return event.

### ISSUE THREAD

**Issue by olssoncharlie** [Lundalogik/lime-elements#1881]
## Current behavior

Using onClick for limel-button-group triggers the event twice. First it returns `1` and then `0`.
The onClick also doesn't return the full event, only the numbers mentioned above.

This occurs both when changing button and clicking on the currently selected one.

1. Create a limel-button-group
2. Add an onClick={(e) => console.log(e)}
3. Click and check console
4. Occurs both when changing button as well as clicking on the current selected one

## Expected behavior

The full event should be available for anyone wanting to create custom behavior onClick. 
One example of this is to trigger a dropdown from a button-group. This is currently impossible due to the event being triggered twice, and not passing the full event to the click handler making it impossible to know whether you clicked on a new button or the currently selected one. 

We want to achieve something similar to this:
![image](https://user-images.githubusercontent.com/25174519/196462006-3baffbd6-2e1a-4023-8661-bdda061631de.png)

## Environment

Lime Go
- **lime-elements** version: 33.12.0 
- Framework used: React 16.8

**Comment by adrianschmidt** [comment]
Yeah, this one is on me. Once upon a time, I thought _"we're making basically drop-in replacements for standard elements like button → limel-button, etc, and then it makes sense that the events we send out replace the native events"_. But, no, it doesn't really. We should have named our events differently 😞

How to fix this is a bit of a tough nut to crack. We can certainly start by adding events with the names we want, alongside the current ones, and deprecate the current ones. Actually removing the current events will be breaking for all consumers (add-ons for example) still using them, which is 😬 to say the least…

Thoughts? @jgroth @BregenzerK @anderssonjohan @olssoncharlie

**Comment by BregenzerK** [comment]
I'm not sure I understand correctly how this connects together. is one click event coming from the original button element and the other one from the limel-button? Aren't we stopping the event from bubbeling in limel-button? I haven't heard from anyone in the addons team that we actually run into a problem around this.

 if we don't have another option than renaming the event, we have to carefully plan it. We're using the onClick listener everywhere, but we can at least prioritize that refactoring. I'm more worried about all the customizations out there

**Comment by adrianschmidt** [comment]
Hmm… this is weird. We don't seem to emit any click event from `limel-button-group`. And we're not stopping the native click event either.

@olssoncharlie Are you sure you're not using `onChange` rather than `onClick`?

**Comment by adrianschmidt** [comment]
Odd. I couldn't reproduce when I tried it in the docs, even when trying with v33.12.0… 🤔

It's soon lunch, and I have promised to help Stavros after lunch. But maybe after that we could have a look at your code together?

**Comment by adrianschmidt** [comment]
Oh, if you could send me a link to your code, that would be great. You can send it in Slack if you don't want to post it here.

**Comment by jgroth** [comment]
Maybe it's an issue when using it together with React? :thinking:

**Comment by olssoncharlie** [comment]
> Maybe it's an issue when using it together with React? 🤔

Could be! Not sure what would cause it tho. Using regular `<input type="radio">` with onClick events returns the full event. I can't really see that our wrapper for Limel in React would cause the behavior, but maybe you can see something I can't :) 
https://github.com/Lundalogik/lime-go-webclient/blob/staging/src/components/Limel/utils/limelReactWrapper.js
 

> Oh, if you could send me a link to your code, that would be great. You can send it in Slack if you don't want to post it here.

I opted for a different solution for now, but just ping me on Slack when you have time and I can show you what I tried earlier.

**Comment by devbymadde** [comment]
Closed due to age. Please reopen if the issue still persists.
